New AZD Project Template - Azure Serverless App with Angular and MSAL

ryanninodizon opened the pull request:
This template project is built using Angular 17-Standalone, Azure Functions, CosmosDB, and APIM, integrated with MSAL (Microsoft Authentication Library) for authentication and authorization. It aims to be a helpful starting point for anyone exploring serverless development and the Microsoft Identity platform.

v-xuto commented:
@ryanninodizon It is recommended to add how to use `azd` to deploy this project in ReadMe, like this [todo template](https://github.com/azure-samples/todo-nodejs-mongo).

ryanninodizon commented:
> @ryanninodizon It is recommended to add how to use `azd` to deploy this project in ReadMe, like this [todo template](https://github.com/azure-samples/todo-nodejs-mongo).

@v-xuto  thank you for your response. The ReadMe is now updated, please check it again:  https://github.com/ryanninodizon/AzureServerlessApp-with-auth-for-dotnet-angular/blob/main/README.md

Thank you and have a nice day.

v-xuto commented:
@hemarina, @ryanninodizon  We have finished this template test, and we filed one issues. Please review.

- https://github.com/ryanninodizon/AzureServerlessApp-with-auth-for-dotnet-angular/issues/1

ryanninodizon commented:
@v-xuto  thank you so much for filing an issue.  There error should be resolved now. see info below. Thanks again and have a nice day.

- updated ReadMe - **Docker** is now included in the list of Prerequisites: https://github.com/ryanninodizon/AzureServerlessApp-with-auth-for-dotnet-angular/blob/main/README.md

- I have added the missing Dockerfile inside frontend-angular project to resolve the issue. Here is the screenshot of my testing.
![azd-deployment](https://github.com/ryanninodizon/AzureServerlessApp-with-auth-for-dotnet-angular/assets/50763950/42960e43-09d9-41ab-bc57-6dcd484b1ad0)

ryanninodizon commented:
@hemarina thank you for your feedback. 

I am little bit confused. what do I need to do next? I need to update the template.json with your suggestions and creatre another PR?

ryanninodizon commented on website/static/templates.json:
```diff
@@ -1897,5 +1897,25 @@
+      "dotnetCsharp",
+      "cosmosdb",
+      "serverless",      
+      "angular",
```
@hemarina msal and angular are now added to tags.tsx

Thank you and have a nice day.

ryanninodizon commented:
> Thank you very much for contributing templates. Just have some small comments.

Thanks @hemarina for the review! I've updated all your suggestions.

v-xuto commented:
@hemarina, @ryanninodizon Currently, we cannot verify whether [issue1](https://github.com/ryanninodizon/AzureServerlessApp-with-auth-for-dotnet-angular/issues/1) has been fixed. Due to some issues with the appinsights service, we are unable to deploy this template successfully. Wait for the resource `application insights` to be created normally, we will retest.

ryanninodizon commented:
@hemarina thank you so much for your quick response. This is noted. 
Have a nice day.

v-xuto commented:
@hemarina [Issue1](https://github.com/ryanninodizon/AzureServerlessApp-with-auth-for-dotnet-angular/issues/1) has been fixed. There are currently no issues with this template. Please review.

ryanninodizon commented:
@hemarina thank you so much for adding/merging my template. Have a nice day.
